rskj-core/src/main/java/co/rsk/net/NodeMessageHandler.java

```diff
@@ -297,202 +281,122 @@ private void processBlockMessage(@Nonnull final MessageChannel sender, @Nonnull
 
         Metrics.processBlockMessage("start", block, sender.getPeerNodeID());
 
-        boolean wasOrphan = !this.blockProcessor.hasBlockInSomeBlockchain(block.getHash());
-
         if (!isValidBlock(block)) {
             logger.trace("Invalid block {} {}", block.getNumber(), block.getShortHash());
             recordEvent(sender, EventType.INVALID_BLOCK);
             return;
         }
 
-        long start = System.nanoTime();
-
+        boolean wasOrphan = !this.blockProcessor.hasBlockInSomeBlockchain(block.getHash());
         BlockProcessResult result = this.blockProcessor.processBlock(sender, block);
 
-        long time = System.nanoTime() - start;
-
-        if (time >= 1000000000)
-            result.logResult(block.getShortHash(), time);
-
         Metrics.processBlockMessage("blockProcessed", block, sender.getPeerNodeID());
 
         recordEvent(sender, EventType.VALID_BLOCK);
+        tryRelayBlock(sender, block, wasOrphan, result);
 
-        long currentTimeMillis = System.currentTimeMillis();
-
-        if (result.anyImportedBestResult())
-            lastImportedBestBlock = currentTimeMillis;
-
-        if (currentTimeMillis - lastImportedBestBlock > WAIT_TIME_ACCEPT_ADVANCED_BLOCKS)
-        {
-            logger.trace("Removed blocks advanced filter in NodeBlockProcessor");
-            this.blockProcessor.acceptAnyBlock();
-            this.receivedMessages.clear();
-        }
+        Metrics.processBlockMessage("finish", block, sender.getPeerNodeID());
+    }
 
+    private void tryRelayBlock(@Nonnull MessageChannel sender, Block block, boolean wasOrphan, BlockProcessResult result) {
         // is new block and it is not orphan, it is in some blockchain
-        if (wasOrphan && result.wasBlockAdded(block) && !this.blockProcessor.isSyncingBlocks())
+        if (!wasOrphan && result.wasBlockAdded(block) && !this.blockProcessor.hasBetterBlockToSync())
```

Comment:
Looks like we had a bug, great catch

Reply:
I read the code and I believe the proper implementation is as said on the comment line